extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java

```diff
@@ -114,6 +122,17 @@ public void updateClasspath(IProject project, IProgressMonitor monitor) throws C
                 new File(jvmBuildTarget.getJavaHome()));
         classpath.add(JavaCore.newContainerEntry(JavaRuntime.newJREContainerPath(vm)));
 
+
+        // Use a map to deduplicate dependencies which point to the same jar.
+        Map<String, IClasspathEntry> dependencyMap = new HashMap<>();
+        for (BuildTarget buildTarget : buildTargets) {
+            boolean isTest = buildTarget.getTags().contains(BuildTargetTag.TEST);
+            DependencyModulesResult dependencyModuleResult = buildServer.buildTargetDependencyModules(
+                    new DependencyModulesParams(Arrays.asList(buildTarget.getId()))).join();
+            addDependencyJars(dependencyMap, dependencyModuleResult, isTest);
```

Comment:
If a dependency is used by both runtime & test build target, then it should not be marked as test. Only those that are exclusively used by test build target need to be marked as test. In current implementation, it doesn't guarantee that.

Reply:
The test build targets are moved to the end of the list at line 90